pkg/operator/ceph/object/spec.go

```diff
@@ -372,7 +372,8 @@ func (c *clusterConfig) makeDaemonContainer(rgwConfig *rgwConfig) (v1.Container,
 				strings.TrimPrefix(generateCephXUser(rgwConfig.ResourceName), "client.")),
 			"--foreground",
 			cephconfig.NewFlag("rgw frontends", fmt.Sprintf("%s %s", rgwFrontendName, c.portString())),
-			cephconfig.NewFlag("host", controller.ContainerEnvVarReference(k8sutil.PodNameEnvVar)),
+			cephconfig.NewFlag("crush location", fmt.Sprintf("%s=%s", "host", controller.ContainerEnvVarReference(k8sutil.NodeNameEnvVar))),
```

Comment:
Should this unit test change belong to #15518?

Reply:
Yes. This belongs to the other PR.  I'll remove it from this PR after testing.